### PR TITLE
Dead code removal

### PR DESCRIPTION
**Pull request by wirthi**
That's as far as I dare to go. Feel free to object if you don't want to see anything removed.

### CODE REVIEW

File: src/main/java/org/truffleruby/core/string/KCode.java [hunk 1]
```diff
@@ -31,73 +31,26 @@
...
-        if ((bits & 48) == 48) {
-            return SJIS;
-        }
-        if ((bits & 32) != 0) {
-            return EUC;
-        }
-
-        return NONE;
-    }
-
```
Comment (chrisseaton) [inline src/main/java/org/truffleruby/core/string/KCode.java]: It's possible this 'K code' logic is needed for passing specs that we have tagged at the moment, and we should try to turn it on. But we can refer back to these commits when we need to do that.

Comment (wirthi) [reply src/main/java/org/truffleruby/core/string/KCode.java]: If you prefer, I can leave it in

Comment (chrisseaton) [reply src/main/java/org/truffleruby/core/string/KCode.java]: No if it’s dead it should be debrided.

File: src/main/java/org/truffleruby/core/string/StringNodes.java [hunk 2]
```diff
@@ -183,10 +183,6 @@ public DynamicObject fromRope(Rope rope) {
             return executeMake(rope, NotProvided.INSTANCE, NotProvided.INSTANCE);
         }
 
-        public DynamicObject fromBuilder(RopeBuilder builder, CodeRange codeRange) {
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/core/string/StringNodes.java]: I'd like to keep this one for now. I have it there so people don't try to solve this problem their own way. It's a complement to the `fromRope` method above it.

File: src/main/java/org/truffleruby/interop/InteropNodes.java [hunk 3]
```diff
@@ -582,10 +582,6 @@ public Object read(TruffleObject receiver, Object identifier,
             return foreignToRubyNode.executeConvert(foreign);
         }
 
-        protected int getCacheLimit() {
-            return getContext().getOptions().INTEROP_READ_CACHE;
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/interop/InteropNodes.java]: It looks like we can remove this from `tool/options.yml`and regenerate `Options` & `OptionsCatalog` with `jt build options`.

File: src/main/java/org/truffleruby/interop/InteropNodes.java [hunk 4]
```diff
@@ -620,10 +616,6 @@ public Object write(TruffleObject receiver, Object identifier, Object value,
             return foreignToRubyNode.executeConvert(foreign);
         }
 
-        protected int getCacheLimit() {
-            return getContext().getOptions().INTEROP_WRITE_CACHE;
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/interop/InteropNodes.java]: It looks like we can remove this from `tool/options.yml`and regenerate `Options` & `OptionsCatalog` with `jt build options`.

File: src/main/java/org/truffleruby/core/string/StringNodes.java (hunk 2 above)
Comment (wirthi) [reply src/main/java/org/truffleruby/core/string/StringNodes.java]: ok, will leave it. Please try to make code like that being used at least from a test if possible.

File: src/main/java/org/truffleruby/interop/InteropNodes.java (hunk 4 above)
Comment (wirthi) [reply src/main/java/org/truffleruby/interop/InteropNodes.java]: thanks, did, and updated the PR.

**Pull request merged**